app/crud/wh_sched.py

```diff
@@ -130,25 +133,222 @@ def verify_scales(cls, values):
+            + "RUN TIMESTAMP_LTZ(9), SUCCESS BOOLEAN, OUTPUT VARIANT)"
+        ).collect()
 
-_WAREHOUSE_SIZE_OPTIONS = [
-    "X-Small",
-    "Small",
-    "Medium",
-    "Large",
-    "X-Large",
-    "2X-Large",
-    "3X-Large",
-    "4X-Large",
-    "5X-Large",
-    "6X-Large",
-    "Medium Snowpark",
-    "Large Snowpark",
-    "X-Large Snowpark",
-    "2X-Large Snowpark",
-    "3X-Large Snowpark",
-    "4X-Large Snowpark",
-]
+
+def describe_warehouse(session: Session, warehouse: str):
+    wh_df = session.sql(f"show warehouses like '{warehouse}'").collect()
+    wh_dict = wh_df[0].as_dict()
+    return WarehouseSchedules(
+        name=warehouse,
+        size=wh_dict["size"],
+        suspend_minutes=int(wh_dict["auto_suspend"] or 0) // 60,
+        resume=wh_dict["auto_resume"],
+        scale_min=wh_dict.get("min_cluster_count", 0),
+        scale_max=wh_dict.get("max_cluster_count", 0),
+        warehouse_mode=wh_dict.get("scaling_policy", "Standard"),
+    )
+
+
+def compare_warehouses(
+    warehouse_now: WarehouseSchedules, warehouse_next: WarehouseSchedules
+):
+    changes = []
+    if (
+        warehouse_now.scale_min == 0
+        or warehouse_next.scale_min == 0
+        or warehouse_now.scale_min == 0
+        or warehouse_next.scale_min == 0
+    ):
+        is_standard = True
+    else:
+        is_standard = False
+    if warehouse_now.size != warehouse_next.size:
+        changes.append(
+            f"WAREHOUSE_SIZE = {_WAREHOUSE_SIZE_OPTIONS[warehouse_next.size]}"
+        )
+        if "Snowpark" in warehouse_next.size and "Snowpark" not in warehouse_now.size:
+            changes.append("WAREHOUSE_TYPE = SNOWPARK-OPTIMIZED")
+        if "Snowpark" not in warehouse_next.size and "Snowpark" in warehouse_now.size:
+            changes.append("WAREHOUSE_TYPE = STANDARD")
+    if warehouse_now.suspend_minutes != warehouse_next.suspend_minutes:
+        changes.append(f"AUTO_SUSPEND = {warehouse_next.suspend_minutes}")
+    if warehouse_now.resume != warehouse_next.resume:
+        changes.append(f"AUTO_RESUME = {warehouse_next.resume}")
+    if warehouse_now.scale_min != warehouse_next.scale_min and not is_standard:
+        changes.append(f"MIN_CLUSTER_COUNT = {warehouse_next.scale_min}")
+    if warehouse_now.scale_max != warehouse_next.scale_max and not is_standard:
+        changes.append(f"MAX_CLUSTER_COUNT = {warehouse_next.scale_max}")
+    if (
+        warehouse_now.warehouse_mode != warehouse_next.warehouse_mode
+        and not is_standard
+        and not warehouse_next.warehouse_mode == "Inherit"
+    ):
+        changes.append(f"SCALING_POLICY = {warehouse_next.warehouse_mode}")
+    return ", ".join(changes)
+
+
+def update_warehouse(warehouse_now, warehouse_next):
+    set_stmts = compare_warehouses(warehouse_now, warehouse_next)
+    if set_stmts:
+        return f"alter warehouse {warehouse_next.name} set {set_stmts}"
+    return ""
+
+
+class WarehouseSchedulesTask:
```

Comment:
I am confused about why this class has to exist. It has no state at all

Reply:
Sorry, I forgot to call this out when publishing the PR. I was struggling to get the mocking to work with the logic as module functions (not sure if that's correct terminology -- as your original draft PR had implemented). I had already wasted a couple of hours fiddling with mocking instead of writing tests. Making this a class made the mocking much easier to set up. I would not normally advocate for product changes just to enable testing, but this felt like an appropriate exception given the time I had already spent.

Reply:
Annoying, but I agree. Mind just putting a comment to that effect in the class pydoc. We can fix it later if we are so inclined

Reply:
of course.